safenode/src/bin/kadclient.rs

```diff
@@ -6,75 +6,30 @@
 // KIND, either express or implied. Please review the Licences for the specific language governing
 // permissions and limitations relating to use of the SAFE Network Software.
 
-use safenode::{
-    client::{Client, ClientEvent, Error as ClientError, Files, WalletClient},
-    log::init_node_logging,
-    protocol::{
-        address::ChunkAddress,
-        wallet::{parse_public_address, DepositWallet, LocalWallet, Wallet},
-    },
-};
+mod cli;
 
-use sn_dbc::{Dbc, Token};
+use self::cli::{cfg_cmds, files_cmds, register_cmds, wallet_cmds, Opt};
 
-use bytes::Bytes;
-use clap::Parser;
-use dirs_next::home_dir;
-use eyre::Result;
-use std::{fs, path::PathBuf, thread, time};
-use tracing::{info, warn};
-use walkdir::WalkDir;
-use xor_name::XorName;
-
-#[derive(Parser, Debug)]
-#[clap(name = "safeclient cli")]
-struct Opt {
-    #[clap(long)]
-    log_dir: Option<PathBuf>,
-    /// The location of the wallet file.
-    #[clap(long)]
-    wallet_dir: Option<PathBuf>,
-    /// Tries to load a hex encoded `Dbc` from the
-    /// given path and deposit it to the wallet.
-    #[clap(long)]
-    deposit: Option<PathBuf>,
-    /// This must be a hex-encoded `PublicAddress`.
-    #[clap(long)]
-    send_to: Option<String>,
-    /// This shall be the number of nanos to send.
-    /// Necessary if the `send_to` argument has been given.
-    #[clap(long)]
-    send_amount: Option<String>,
-
-    #[clap(long)]
-    upload_chunks: Option<PathBuf>,
-
-    #[clap(long)]
-    get_chunk: Option<String>,
+use crate::cli::SubCmd;
 
-    #[clap(long)]
-    create_register: Option<String>,
+use safenode::client::{Client, ClientEvent};
 
-    #[clap(long)]
-    entry: Option<String>,
-
-    #[clap(long)]
-    query_register: Vec<String>,
-
-    #[clap(long)]
-    repeated: bool,
-}
+use clap::Parser;
+use eyre::Result;
+use tracing::info;
 
 #[tokio::main]
 async fn main() -> Result<()> {
     let opt = Opt::parse();
-    let _log_appender_guard = init_node_logging(&opt.log_dir)?;
+
+    if let SubCmd::Cfg(cmds) = &opt.cmd {
+        cfg_cmds(cmds).await?;
```

Comment:
This is also called later on in the `match` statement.

Reply:
Yep. I had to have it the first thing we do, since logs need to be enabled from start.

But then I didn't want to not handle it further down. Shall I wild card that one below? That's less desireable I feel? (Wild carding in general, *potential* double handle is better then?)

Reply:
Fixed!